react-hot-button-component/src/index.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { clickCount: 0 };
+
+    // This binding is necessary to make `this` work in the callback
+    this.countClick = this.countClick.bind(this);
+  }
+
+  countClick() {
+    this.setState({ clickCount: this.state.clickCount + 1 });
+  }
+
+  render() {
+    let className = 'white';
+    const state = this.state.clickCount;
+
+    if (state >= 3 && state < 6) {
```

Comment:
I would move this logic into a method such as `getColorClass` or something like that and have it return the proper classname so you don't have to keep it all in the render method.